libcudacxx/include/cuda/std/__cuda/semaphore.h

```diff
@@ -23,23 +24,147 @@
 
 _LIBCUDACXX_BEGIN_NAMESPACE_CUDA
 
+void simple_backoff() {
+  int wait_iterations = 1;
+  volatile int ctr = 0;
+  const int max_iterations = 1024;
+
+  for (int i = 0; i < wait_iterations; ++i)
+  {
+    ctr++;
+  }
+  wait_iterations = (wait_iterations * 2 >= max_iterations) ? max_iterations : wait_iterations * 2;
+}
```

Comment:
Why do we need a new backoff mechanism here instead of relying on `atomic.wait`?

Reply:
TBH I didn't know that was available. I'm assuming everything in `cccl/libcudacxx/include/cuda/std/atomic`? I'll do that now, this isn't needed